File: kotlin/src/main/kotlin/com/trendyol/posintegrationapi/service/est/model/EstAuthResponse.kt
package com.trendyol.posintegrationapi.service.est.model

class EstAuthResponse {
    var orderId: String? = null
    var groupId: String? = null
    var response: String? = null
    var authCode: String? = null
    var hostRefNum: String? = null
    var procReturnCode: String? = null
    var transactionId: String? = null
    var errMsg: String? = null
    var rawBody: String? = null
    var isSuccess: Boolean = "Approved" == response
}

File: golang/service/est/model/est_auth_response.go
package model

type EstAuthResponse struct {
	OrderId        string
	GroupId        string
	Response       string
	AuthCode       string
	HostRefNum     string
	ProcReturnCode string
	TransactionId  string
	ErrMsg         string
	RawBody        string
}

func (e *EstAuthResponse) IsSuccess() bool {
	return e.Response == "Approved"
}
